main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/sirupsen/logrus"
+	"github.com/skynetlabs/pinner/api"
+	"github.com/skynetlabs/pinner/conf"
+	"github.com/skynetlabs/pinner/database"
+	"gitlab.com/NebulousLabs/errors"
+)
+
+func main() {
+	// Load the configuration from the environment and the local .env file.
+	err := conf.LoadConfiguration()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Initialise the global context and logger. These will be used throughout
+	// the service. Once the context is closed, any background threads will
+	// wind themselves down.
+	ctx := context.Background()
+	logger := logrus.New()
+	logLevel, err := logrus.ParseLevel(conf.LogLevel)
+	if err != nil {
+		logLevel = logrus.InfoLevel
+	}
+	logger.SetLevel(logLevel)
+
+	// Initialised the database connection.
+	dbCreds := database.DBCredentials{
+		User:     conf.DBUser,
+		Password: conf.DBPassword,
+		Host:     conf.DBHost,
+		Port:     conf.DBPort,
+	}
+	db, err := database.New(ctx, dbCreds, logger)
+	if err != nil {
+		log.Fatal(errors.AddContext(err, "failed to connect to the db"))
+	}
+
+	// Initialise the server.
+	server, err := api.New(db, logger)
+	if err != nil {
+		log.Fatal(errors.AddContext(err, "failed to build the api"))
+	}
+
+	log.Fatal(server.ListenAndServe(4000))
```

Comment:
I'm not sure it's worth it, tbh. We're running this in docker, which means that we don't face a risk of colliding port numbers. And if we want to expose it on another port we can always do that in the docker-compose file.

Also, we don't have that configurable in either accounts, blocker, or malware-scanner.

If we think this would be useful, I can add it in a F/U.

Reply:
ah this is the internal port. I was thinking with was going to be the docker referenced in the dockerfile.

nm then.